src/libs/API/commands/ValidateEmailCommand.js

```diff
@@ -0,0 +1,78 @@
+import lodashGet from 'lodash/get';
+import Onyx from 'react-native-onyx';
+import _ from 'underscore';
+import ONYXKEYS from '../../../ONYXKEYS';
+import * as User from '../../actions/User';
+import BaseAPICommandBlocking from './BaseAPICommandBlocking';
+import CONST from '../../../CONST';
+import Navigation from '../../Navigation/Navigation';
+import ROUTES from '../../../ROUTES';
+
+let isLoggedIn = nul;
+Onyx.connect({
+    key: ONYXKEYS.SESSION,
+    callback: (val) => {
+        const sessionAuthToken = lodashGet(val, 'authToken', '');
+        isLoggedIn = !_.isEmpty(sessionAuthToken);
+    },
+});
+
+let currentlyViewedReportID = '';
+Onyx.connect({
+    key: ONYXKEYS.CURRENTLY_VIEWED_REPORTID,
+    callback: val => currentlyViewedReportID = val || '',
+});
+
+export default class extends BaseAPICommandBlocking {
+    commandName = 'ValidateEmail';
+
+    commandParameters = {};
+
+    requiredParameters = ['accountID', 'validateCode'];
+
+    startBlocking() {
+        super.startBlocking();
+        Onyx.merge(ONYXKEYS.ACCOUNT, {...CONST.DEFAULT_ACCOUNT_DATA});
+    }
+
+    finishBlocking() {
+        super.finishBlocking();
+
+        const redirectRoute = isLoggedIn ? ROUTES.getReportRoute(currentlyViewedReportID) : ROUTES.HOME;
+
+        Navigation.navigate(redirectRoute);
+    }
+
+    processDataChanged(changedData) {
+        // @TODO This data shouldn't have anything except the email
+        const {email} = changedData;
+
+        if (isLoggedIn) {
+            // @TODO This data should ould be returned in the changedData payload
+            User.getUserDetails();
+        } else {
+            // Let the user know we've successfully validated their login
+            const success = lodashGet(changedData, 'message', `Your secondary login ${email} has been validated.`);
+            Onyx.merge(ONYXKEYS.ACCOUNT, {success});
+        }
+    }
+
+    /**
+     * @param {Number} accountID
+     * @param {String} validateCode
+     */
+    makeRequest(accountID, validateCode) {
+        super.makeRequest({accountID, validateCode});
```

Comment:
**when defining the action:** 

```js
/**
 * Validates a login given an accountID and validation code
 *
 * @param {Number} accountID
 * @param {String} validateCode
 */
function validateLogin(accountID, validateCode) {
    return Command_ValidateEmail.request({accountID, validateCode});
}
```

Reply:
Yeah, we could do that. I'm usually against always passing objects of params except for when necessary, primarily because of IDE hints and how helpful they are for people learning the code.

Reply:
valid concern for sure. we could do fancy JSDoc things to get around that also but maybe too fancy 💅

Reply:
I think if we're gonna use classes we should make the method signatures the same, otherwise why are we using classes?

Reply:
>the way to actually use classes in this design would be to make the signatures the same and not even have this method defined in this class

Do you just mean something like this?

```js
class ValidateEmail extends NetworkAction {
    request(accountID, validateCode) {
        // Where requestWithParams is the parent method
        this.requestWithParams({accountID, validateCode});
    }
}
```

Reply:
Not quite:
- remove makeRequest from ValidateEmailCommand
- update the params passed to makeRequest here to be an object `(new ValidateEmailCommand()).makeRequest({accountID, validateCode});`